public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -56,6 +56,36 @@ export function partitionTimeRange(
   });
 }
 
+/**
+ * Based in the state of the current response, if any, adjust target parameters such as `maxLines`.
+ * For `maxLines`, we will update it as `maxLines - current amount of lines`.
+ * At the end, we will filter the targets that don't need to be executed in the next request batch,
+ * becasue, for example, the `maxLines` have been reached.
+ */
+
+function adjustTargetsFromResponseState(targets: LokiQuery[], response: DataQueryResponse | null): LokiQuery[] {
+  if (!response) {
+    return targets;
+  }
+
+  return targets
+    .map((target) => {
```

Comment:
i recommend adding an
```js
if(!isLogsQuery(target.expr)) {
    return false
}
```
at the beginning of this function,

because we do not want to do this for metric-queries,
and we may have `maxLines` filled out for a metric-query.

Reply:
Ohh, good catch.

Reply:
Done.